### PR TITLE
chore: Remove error duplicates and standarize exceptions (part 1)

### PR DESCRIPTION
**Pull request by enikonovad**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [ ] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary
This is a sample of the proposed structure to follow in the project when it comes to the exception handling and logging. The aim of the change is to remove the error duplicates and use appropriate logger level for logger messages.

The proposal consists of several parts:

### Logger level standardisation:
When to use which logging level:
- Info (logger.info): Provide information on successful execution. Confirm that things are working as expected.
- Warning (logger.warning): Issue a warning to the user that there is a problem that can lead to some issues or bugs. However, the problem does not halt the execution and operation proceeds.
- Error (logger.error): Indicates to a user that there is critical problem and execution cannot proceed. Such problem must end the execution and exit with appropriate code.

### Exception handling and logging
- `Logger.error` is no longer allowed anywhere but in the most outer level.
- `Logger.warning` and `Logger.info` allowed anywhere but the execution after `logger.warning` **must** proceed.
- Traceback will only be printed if command is executed in the verbose mode.
- Error exit codes are determined by exception handler in outer most level based on the exception type.
- Handling of inner exceptions - the inner exceptions should not be caught unless there is need to provide more information to the user on what went wrong. In this case, an appropriate instance of `TrestleError` should be raised.
  Example: 
```
except SomeException as e:
    raise TrestleError(f'Something went wrong during SSP generation: {e}')
```

- Each `_run()` function should be wrapped in `try-except` block.
- The outer most exception should be handled via `handle_generic_command_exception` with custom message and its output should be returned by `_run()`
- Raise exception rather than `return` command code manually.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

**Comment by fsuits** [comment]
Looks good.  Note there is also the trace.log() mechanism for things intended only in -vv mode (extra verbose)
Should the generic handler be returned in the exceptions or raised?  You show them being returned.

**Comment by enikonovad** [comment]
@fsuits this is a first part of the error messaging refactoring, the rest will come later as it already creates lots of merge conflicts. Please review when have a chance

File: tests/trestle/core/commands/transform_oscal_test.py [hunk 1]
```diff
@@ -97,9 +100,10 @@ def test_ssp_transform(tmp_trestle_dir: pathlib.Path) -> None:
     rc = transform_oscal._run(args)
     assert rc == 0
 
-    assert transform_oscal.transform(
-        tmp_trestle_dir, const.MODEL_TYPE_SSP, const.FILTER_BY_PROFILE, 'bar', 'myout', 'myprof', False, None
-    ) == 1
+    with pytest.raises(TrestleError):
+        transform_oscal.transform(
+            tmp_trestle_dir, const.MODEL_TYPE_SSP, const.FILTER_BY_PROFILE, 'bar', 'myout', 'myprof', False, None
+        )
```
Comment (fsuits) [inline tests/trestle/core/commands/transform_oscal_test.py:106]: transform_oscal has been removed so just accept removal after merge conflict.

File: trestle/core/commands/author/folders.py [hunk 2]
```diff
@@ -150,46 +143,37 @@ def template_validate(
...
-                continue
-            elif template_file.suffix.lower() == '.md':
-                if not readme_validate and template_file.name == 'readme.md':
-                    logger.error('Template directory contains a readme.md file and readme validation is off.')
-                    return CmdReturnCodes.COMMAND_ERROR.value
-                try:
+            try:
+                if not file_utils.is_local_and_visible(template_file):
+                    continue
+                elif template_file.is_dir():
```
Comment (fsuits) [inline trestle/core/commands/author/folders.py:156]: surprised to see a fairly low level try/except here - but i guess it's ok?

Comment (enikonovad) [reply trestle/core/commands/author/folders.py:156]: I think in this case it is okay, as it provides some specific information to the user, which only this function knows (i.e. which template file is faulty)

File: tests/trestle/core/commands/transform_oscal_test.py (hunk 1 above)
Comment (enikonovad) [reply tests/trestle/core/commands/transform_oscal_test.py:106]: Done

**Review by fsuits — APPROVED** [review]
LGTM

**Pull request merged**